types/bitcoinjs-lib/index.d.ts

```diff
@@ -1,114 +1,234 @@
-// Type definitions for bitcoinjs-lib 2.3
-// Project: https://github.com/bitcoinjs/bitcoinjs-lib#readme
-// Definitions by: Mohamed Hegazy <https://github.com/mhegazy>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/** Declaration file initially generated by dts-gen for bitcoinjs-lib 3.0.3*/
+
+// Definitions by: Daniel <https://github.com/dlebrecht>.
+/* modified by dlebrecht based on the flow version found at https://github.com/flowtype/flow-typed/blob/master/definitions/npm/bitcoinjs-lib_v2.x.x/flow_v0.17.x-/bitcoinjs-lib_v2.x.x.js */
+
+
+export type Utxo = {
+  address: string;
+  amount: number;
+  scriptPubKey: Buffer;
+  txid: string;
+  vout: number;
+}
+
+export type Output = {
+    script: Buffer;
+    value: number;
+};
+
+export type Input = {
+  script: Buffer;
+  hash: Buffer;
+  index: number;
+  sequence: number;
+};
+
+export type Network = {
+  bip32: {
+    public: number;
+    private: number;
+  };
+  messagePrefix: string;
+  pubKeyHash: number;
+  scriptHash: number;
+  wif: number;
+}
+
 
 export class Block {
     constructor();
+
+    byteLength(headersOnly: any): any;
+
     checkMerkleRoot(): any;
+
     checkProofOfWork(): any;
-    getHash(): any;
-    getId(): any;
+
+    getHash(): Buffer;
+
+    getId(): string;
+
     getUTCDate(): any;
-    toBuffer(headersOnly: any): any;
-    toHex(headersOnly: any): any;
+
+    toBuffer(headersOnly?: boolean): Buffer;
+
+    toHex(headersOnly?: boolean): string;
+
     static calculateMerkleRoot(transactions: any): any;
```

Comment:
Generally, `transactions` should have the type `Transaction[]` and return `Buffer`.
Realistically, `transactions` should have the type `{ getHash(): Buffer; }[]` and return `Buffer`.